src/prog_translator.cc

```diff
@@ -55,9 +57,22 @@ void ProgTranslator::Init() {
   ctx_.m = CoinPackedMatrix(false, 0, 0);
 }
 
+void ProgTranslator::CheckPref(double pref) {
+  if (pref <= 0) {
```

Comment:
@gidden - maybe then, the exception here should say ``"resource exchange implementation is not working right"`` because this would only fire if there was a bug in cyclus right?